src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -74,32 +84,121 @@ protected void doGet(HttpServletRequest request,HttpServletResponse response){
 		}
 		
 	}
+
+	private void checkIfACourseReportThreadIsDone(HttpServletRequest request,HttpServletResponse response) {
+		M_log.debug("Starting polling... Thread size: "+courseReportReqThreads.size());
+		try {
+			String thrdId = request.getParameter(THREAD_ID);
+			long threadId = Long.parseLong(thrdId);
+			KeySetView<Thread, CourseReportTask> keySet = courseReportReqThreads.keySet();
+			for (Thread aCourseReportThread : keySet) {
+				if(aCourseReportThread.getId()==threadId) {
+					checkIfThreadAlive(response, aCourseReportThread);
+					break;
+				}
+
```

Comment:
What happens if the value of `threadId` isn't found in any of the threads?  Does it fail silently?  It should report an error here.